enhancements/network/user-space-pod-iface-and-api-library.md

```diff
@@ -0,0 +1,382 @@
+needs to be passed to the pod. Below are types of data that need to be passed
+into the pod:
+
+- Multiple interfaces:
+  - With multiple interfaces now in a pod, the pod needs context on which
+  interface is intended for what purpose.
+
+- Interface is used by a user space networking application in the pod:
+  - When the container workload performs packet processing in user space, for
+  example a DPDK application, some of the DPDK poll mode drivers (logic that
+  polls the interface for packets) require different kernel drivers in order to
+  work properly. The interfaces are moved over to these drivers
+  (uio_pci_generic, igb_uio, or vfio-pci) before being assigned to a pod. These
+  drivers do not have a mechanism to assign an IP address or MAC address from
+  host (i.e. from a CNI) and such can not be read from within a pod. So for
+  user space networking applications, another mechanism needs to used to pass
+  IP address and MAC address values to a pod.
+  - In addition to data typically passed via the kernel interface (IP address
+  and MAC address), user space networking applications require additional
+  information to function properly. Examples of some of the data include:
+    - CPUs allocated to the pod
+    - Number of hugepages dedicated to the pod
+
+- User space interface is injected into the pod:
+  - A user space interface, like a [vhost-user](#vhost-user) (not supported by
+  OCP) or a [vDPA interface](#vdpa-interface), which uses the virtio protocol
+  for data plane negotiation (support expect in future OCP release, around 4.8
+  or 4.9 time frame), requires additional information like Unix socket file and
+  virtio mode (master or client). Additional data needed by these types of
+  interfaces is not needed by OCP immediately, but must be considered during the
+  overall design.
+
+This enhancement has two components:
+
+- How the data should be passed to the pod?
+- How to ease the consumption of the data in the pod?
+
+## Motivation
+
+Various system resources are allocated to run high performance network
+applications inside a container such as CPU cores, dedicated memory (hugepages),
+network devices, etc. However, these resources are presented differently inside
+the container and some of them are not visible to applications without
+explicitly injecting to the container.
+
+### Goals
+
+- Define what data needs to be passed from Kubernetes to a user space
+networking application to enable the application to run properly.
+
+- Define how this data is passed from Kubernetes to a user space networking
+application.
+
+- Ease the consumption of the data that was passed into a user space networking
+application by providing a library that both provides an example of how the
+data can be collected and if desired, integrated directly into the application. 
+
+- Ensure customers using SR-IOV can run user space networking applications in
+their deployment.
```

Comment:
(I was just using "ensure customers using SR-IOV..." as an example of "sometimes the bullet points are more focused on the customer goal than on the development details". I wasn't necessarily suggesting we actually _need_ to have a bullet point about customers using SR-IOV. Feel free to remove this if you don't think it adds any important perspective to the doc.)

Reply:
Yes, I realized that you were just giving an example of possible items in bullet form. But I think it added value. It was the catch all, to make sure DPDK applications can be run and to add any additional fields that I missed in the design. I could do all this work I mention and it still doesn't work (I think I most covered though). I'll remove if you don't like though.